src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/CreateCalendarEvent/CreateCalendarEventCommandValidator.cs

```diff
@@ -0,0 +1,246 @@
+    public const string LongitudeMustBeValid = "longitude must be between -180 and 180";
+    public const string EventLinkMustBeValid = "eventLink must be a valid url";
+    public const string EventLinkMustBeEmpty = "eventLink must be empty when event format is InPerson";
+    public const string EventLinkMustNotExceedLength = "eventLink must not be greater than 2000 characters long";
+    public const string ContactNameMustNotBeEmpty = "contactName must have a value";
+    public const string ContactNameMustNotExceedLength = "contactName must not be greater than 200 characters long";
+    public const string ContactEmailMustNotBeEmpty = "contactEmail must have a value";
+    public const string ContactEmailMustNotExceedLength = "contactEmail must not be greater than 256 characters long";
+    public const string ContactEmailMustBeValid = "contactEmail must be a valid email format";
+    public const string PlannedAttendeesMustNotBeEmpty = "plannedAttendees must have a value";
+    public const string PlannedAttendeesMustBeValid = "plannedAttendees must be  between 1 to 1000000";
+    public const string RequestedByMemberIdMustNotBeEmpty = "requestedByMemberId must have a value";
+    public const string RequestedByMemberIdMustBeAdmin = "requestedByMemberId must be an active, admin member or regional chair";
+
+    public CreateCalendarEventCommandValidator(
+        ICalendarsReadRepository calendarsReadRepository,
+        IRegionsReadRepository regionsReadRepository,
+        IMembersReadRepository membersReadRepository)
+    {
+        RuleLevelCascadeMode = CascadeMode.Stop;
+
+        RuleFor(c => c.AdminMemberId)
+            .NotEmpty()
+            .WithMessage(RequestedByMemberIdMustNotBeEmpty)
+            .MustAsync(async (memberId, cancellationToken) =>
+            {
+                var member = await membersReadRepository.GetMember(memberId);
+                return
+                    member != null &&
+                    member!.Status == MembershipStatusType.Live.ToString() &&
+                    (member.UserType == UserType.Admin.ToString() || member.IsRegionalChair.GetValueOrDefault());
+            })
+            .WithMessage(RequestedByMemberIdMustBeAdmin)
+            .DependentRules(() =>
+            {
+                RuleFor(c => c.CalendarId)
+                    .GreaterThan(0)
+                    .WithMessage(CalendarTypeIdMustNotBeEmpty)
+                    .MustAsync(async (id, cancellationToken) =>
+                    {
+                        var calendarTypes = await calendarsReadRepository.GetAllCalendars(cancellationToken);
+                        return calendarTypes.Exists(c => c.Id == id);
+                    })
+                    .WithMessage(CalendarTypeIdMustBeValid);
+
+                RuleFor(c => c.EventFormat)
+                    .NotEmpty()
+                    .WithMessage(EventFormatMustNotBeEmpty);
```

Comment:
need to check this is a valid EventFormat Online, InPerson, Hybrid

Reply:
The property type is Enum so it will be one of those values or null

Reply:
I tried rubbish value in swagger to see what happens, the model binder fails as it cannot convert into the enum and returns 400 for us :-)